### PR TITLE
Explicitly capture IxD specs and requirements within repo

### PR DESCRIPTION
**Pull request by fredvisser**
# Pull Request

## 🤨 Rationale

Goals:
- Capture IxD specs and requirements within repo
- Continue to formalize the UX process used to create Nimble components

Essentially, we want to make sure that we have a structure for capturing the essential component requirements in a IxD work item early in the component development process. After we have clarity on the component scope through the reviewed IxD spec, we can begin work on the visual design and technical spec (and build the component).

Nimble Spec Process-3

## 👩‍💻 Implementation

- Created new work item templates for Visual Design and for Integration Design (IxD)
- Created new IxD spec templates
- Updated contributing documentation to point to new templates

## 🧪 Testing

Minimal

## ✅ Checklist

- [x] I have updated the project documentation to reflect my changes or determined no changes are needed.

### CODE REVIEW

File: specs/README.md
Comment (mollykreis) [inline specs/README.md:10]: While not new with your change, I think the word "other" in this line is unnecessary/confusing. I think this should just be

```suggestion
   - If proposing architecture or infrastructure changes, use the [high-level design template](/specs/templates/high-level-design.md) and place the spec at `/specs/title-of-your-proposal/README.md`.
```

File: .github/ISSUE_TEMPLATE/ixd_user_story.md
Comment (mollykreis) [inline .github/ISSUE_TEMPLATE/ixd_user_story.md:2]: I'm not sure if we need to document this somewhere, but I'm still a bit fuzzy on who/when this item gets created. My guess is that this is the workflow:

- Someone creates an issue for a new component
- The implementer of the new component creates an IxD issue with details around the requirements
- The interaction designer creates the issue for the visual design once the interaction design is complete

Some questions:
- Is that general workflow correct?
- Who is ultimately defining all the requirements of a component and getting approval for those? Some PO? The implementer? The interaction designer?
- Does this mean we need to plan components farther in advance than we do now? For example, the current workflow is Design -> Implementation Spec -> Implementation, but it sounds like maybe it now is Requirements Gathering (done by implementer??) -> IxD Design -> Visual Design -> Implementation Spec -> Implementation.

File: specs/templates/component-interaction-design.md
Comment (jattasNI) [inline specs/templates/component-interaction-design.md:1]: Typically interaction designs are better captured in XD/Figma than in a markdown document. Should we update this template to make it clear that it's ok for the bulk of the design to live elsewhere? For example, change "Full description of the proposed interaction design" to "Link to design in XD or Figma and provide any additional context."

Do the other sections even provide value? For example, we don't usually see alternative designs called out in IxD specs. Maybe this whole template could go away and be replaced with a link from the IxD user story to the XD / Figma spec.

File: .github/ISSUE_TEMPLATE/ixd_user_story.md
Comment (jattasNI) [inline .github/ISSUE_TEMPLATE/ixd_user_story.md:14]: Do you envision we would have an IxD user story as a task linked from a top-level "new component" / "feature" issue? If so I worry about collecting requirements and prior art in both places (in the past we've often put this info in the "Problem to Solve" and "Proposed Solution" sections of the new_component feature. I think we should pick one and delete the relevant section from the other place.

File: .github/ISSUE_TEMPLATE/ixd_user_story.md
Comment (jattasNI) [inline .github/ISSUE_TEMPLATE/ixd_user_story.md:35]: Should we add a bullet here describing the handoff to dev? e.g. "IxD Spec presented to feature lead or dev team"

File: .github/ISSUE_TEMPLATE/visual_design_user_story.md
Comment (jattasNI) [inline .github/ISSUE_TEMPLATE/visual_design_user_story.md:17]: If we decide to capture "handoff to dev" as a task for the IxD spec we should do the same for the VxD spec here.

**Comment by rajsite** [comment]
@fredvisser looks like there is some pending feedback, if this is not being actively worked on at the moment can we move it to draft?

**Comment by leslieab** [comment]
This is how I think everything should work, sorry this is my first time responding to a PR and idk if I'm doing it wrong.

**Issues**
The development component/feature issue should include all relevant information, including request background (if applicable). Unless there is existing documentation for both interaction and visual design, do not include links to design work under "proposed solutions". If design has not been documented and/or reviewed, an IxD issue and a ViD issue should be created and linked/included under the development issue task list.

IxD and ViD issues should link back to the development issue and have sections for "proposed solutions", "requirements", and "tasks".
- "Proposed solutions" should link to non-reviewed and other existing assets and documentation.
- "Requirements" should be thought of as what information/assets are needed from design for implementation. For IxD, this should be a list of what the user needs to be able to do. For ViD, this is likely the variations, states, icons, tokens, etc. that the component needs to have.
- "Tasks" should be "design submitted" (with document link in comment or under "proposed solutions") and "design reviewed" (with comments addressed asynchronously and/or after a review meeting).

**Templates**
The "Nimble Component Name [Template]" background section presumably links to the development issue, the interaction design spec, and the visual design spec. I don't see a reason why the "Nimble Component Name Interaction Design [Template]" shouldn't be consistent with the development spec:
1. Overview (_The name of the component, along with high-level description of what the design includes._)
- Background (link to UX issue, link to implementation specification)
3. Design (The headings and content in this section will differ)
4. Open issues (_Highlight any open questions for discussion during the spec PR. Before the spec is approved these should typically be resolved with the answers being incorporated in the spec document._)

An interaction design specification will either be low-fidelity or high-fidelity. Low-fidelity will be used for "non-project work", i.e. simple component that don't require a long-term interaction design time commitment and can be implemented accurately in XD/Figma and code with a high level of detail. The expectation is that they are created in a day or less, and are primarily text-based, with images included as needed/as requested. High-fidelity will be used for "project work" that requires a large amount of interaction design production with very detailed documentation (more in line with what we currently refer to as an interaction design specification).

File: .github/ISSUE_TEMPLATE/ixd_user_story.md
Comment (fredvisser) [reply .github/ISSUE_TEMPLATE/ixd_user_story.md:2]: > Wait, so is the expectation that I create a ViD issue?

No, but let me update the diagram to make that clear – 
image

I'll create both IxD and Visual Design work items, and we'll keep them separate; using the new component request issue/work item to capture any shared information/context. So, if the IxD requirements are minimal, that's completely fine. This way we can delay the creation of the visual design work items until it's clear that we need them, and the requirements are clear.

File: .github/ISSUE_TEMPLATE/ixd_user_story.md
Comment (fredvisser) [reply .github/ISSUE_TEMPLATE/ixd_user_story.md:14]: Yes - I expect this IxD user story to be linked to the top-level feature issue, but I want to be super crisp in the handoff to IxD, so I want to reiterate the key requirements here. I can see the top-level feature being a good forum for brainstorming and discussion, and would be more work to edit down to the core requirements than it is to frame them anew here.

I'll also add a non-requirements section, because it's easy for earlier discussions to create implicit requirements.

Comment (fredvisser) [reply .github/ISSUE_TEMPLATE/ixd_user_story.md:14]: …but I will remove the Prior Art section, as it's better discussed in the top-level feature.

File: .github/ISSUE_TEMPLATE/ixd_user_story.md
Comment (fredvisser) [reply .github/ISSUE_TEMPLATE/ixd_user_story.md:35]: I kept it generic `IxD Spec reviewed by dev team`, but let me know if you'd like to be explicit about saying it should be presented to the team in a team meeting, in addition to reviewing and approving the IxD spec.

File: specs/templates/component-interaction-design.md
Comment (fredvisser) [reply specs/templates/component-interaction-design.md:1]: > It probably doesn't need the "alternative implementations/designs" section.

Removed

>  For example, change "Full description of the proposed interaction design" to "Link to design in XD or Figma and provide any additional context."

Kept both

**Comment by fredvisser** [comment]
> The development component/feature issue should include all relevant information, including request background (if applicable). Unless there is existing documentation for both interaction and visual design, do not include links to design work under "proposed solutions". If design has not been documented and/or reviewed, an IxD issue and a ViD issue should be created and linked/included under the development issue task list.

Good point - removed the example link to visual design in the `new_component.md` template
> 
> IxD and ViD issues should link back to the development issue and have sections for "proposed solutions", "requirements", and "tasks".
> 
> "Proposed solutions" should link to non-reviewed and other existing assets and documentation.

Instead of duplicating the "Proposed solutions" across the New Component issue and the IxD Issue, I'd like to keep that info in the New Component issue, and rely on the requirements to communicate the design constraints. I worry that "proposed solutions" will include implied requirements and I'd like to be as explicit as possible.

> "Requirements" should be thought of as what information/assets are needed from design for implementation. For IxD, this should be a list of what the user needs to be able to do. For ViD, this is likely the variations, states, icons, tokens, etc. that the component needs to have.

Good point – I've added `Core Requirements` and `Non-requirement` sections to each of the ViD and IxD specs.

> "Tasks" should be "design submitted" (with document link in comment or under "proposed solutions") and "design reviewed" (with comments addressed asynchronously and/or after a review meeting).

Done

> Templates
> The "Nimble Component Name [Template]" background section presumably links to the development issue, the interaction design spec, and the visual design spec. I don't see a reason why the "Nimble Component Name Interaction Design [Template]" shouldn't be consistent with the development spec:
> 
> Overview (The name of the component, along with high-level description of what the design includes.)
> Background (link to UX issue, link to implementation specification)
> Design (The headings and content in this section will differ)
> Open issues (Highlight any open questions for discussion during the spec PR. Before the spec is approved these should typically be resolved with the answers being incorporated in the spec document.)
> An interaction design specification will either be low-fidelity or high-fidelity. Low-fidelity will be used for "non-project work", i.e. simple component that don't require a long-term interaction design time commitment and can be implemented accurately in XD/Figma and code with a high level of detail. The expectation is that they are created in a day or less, and are primarily text-based, with images included as needed/as requested. High-fidelity will be used for "project work" that requires a large amount of interaction design production with very detailed documentation (more in line with what we currently refer to as an interaction design specification).

Sounds good - I believe I've made them consistent… please check if I succeeded

File: specs/templates/component-interaction-design.md
Comment (leslieab) [inline specs/templates/component-interaction-design.md]:
```suggestion
## Overview
```

File: specs/templates/component-interaction-design.md
Comment (leslieab) [inline specs/templates/component-interaction-design.md]:
```suggestion
*The name of the component, along with high-level description of what the design includes.*
```

File: specs/templates/component-interaction-design.md
Comment (leslieab) [inline specs/templates/component-interaction-design.md]:
```suggestion
### Background
```

File: specs/templates/component-interaction-design.md
Comment (leslieab) [inline specs/templates/component-interaction-design.md]:
```suggestion
*Relevant historical or background information; link to UX issue implementation specification, etc.*
```

File: specs/templates/component-interaction-design.md
Comment (leslieab) [inline specs/templates/component-interaction-design.md]:
```suggestion
## Design
```

File: specs/templates/component-interaction-design.md
Comment (leslieab) [inline specs/templates/component-interaction-design.md]: Removed (if this information is necessary, I will add it as needed)

File: specs/templates/component-interaction-design.md
Comment (leslieab) [inline specs/templates/component-interaction-design.md:21]:
```suggestion
*Highlight any open questions for discussion during the spec PR. Before the spec is approved these should typically be resolved with the answers being incorporated in the spec document.*
```

**Review by leslieab — COMMENTED** [review]
This is what I think the spec should look like

File: specs/templates/component-interaction-design.md
Comment (fredvisser) [reply specs/templates/component-interaction-design.md]: I currently have the following, which works for me… let me know if you disagree.

```
*Relevant historical or background information*
*Link to Interaction Design issue*
*Link to relevant work items, related existing issues, etc.*
```

File: .github/ISSUE_TEMPLATE/ixd_user_story.md
Comment (jattasNI) [inline .github/ISSUE_TEMPLATE/ixd_user_story.md:2]: We can't use the same emoji for interaction design as we did for visual design! My suggestion is for a boring office worker instead but you or @leslieab has free reign to pick something else. 🧑‍🎤 and 🤦 were leading contenders for me.

```suggestion
name: 🧑‍💼 IxD User story
```

File: .github/ISSUE_TEMPLATE/visual_design_user_story.md
Comment (jattasNI) [inline .github/ISSUE_TEMPLATE/visual_design_user_story.md:5]: The inconsistent usage of "interaction design", "IxD", and "visual design" bugs me a bit. I'd propose either:
1. spell it out everywhere and stop using "IxD"
2. spell it out in the titles and names of things but use "IxD" and "ViD" (I saw Leslie using this term) for issue labels or other places where an abbreviation is appropriate.

**Review by jattasNI — APPROVED** [review]

File: specs/templates/component-interaction-design.md
Comment (fredvisser) [reply specs/templates/component-interaction-design.md:19]: Great point – I've added the following section…

--------

## Usage

*How, when, and where should you use this component?*
*Compare this component with other related components. How does it differ?*
*How should you decide between related components?*

File: .github/ISSUE_TEMPLATE/visual_design_user_story.md
Comment (fredvisser) [reply .github/ISSUE_TEMPLATE/visual_design_user_story.md]: I agree - Current strategy - use full names everywhere, except in the issue tags. However, still define the short names in a few places so that when people inevitably use them, it's easy to find the definition.

**Pull request merged**
